continuous-integration/reference-architectures/README.md

```diff
@@ -1,3 +1,256 @@
+development and devops communities by publishing the files to one or more artifact repositories, 
+such as [JFrog Artifactory](https://jfrog.com/artifactory/). Packaged binaries are versioned, 
+checksummed and do not change after release -- core features of reliable and reproducible CI 
+software delivery.
+
+#### Continuous Deployment System
+- **Orchestrate** (optional)  
+The orchestration process downloads files and prepares target servers to launch. It moves compiled 
+binaries and properties into place and configures required services on destination systems. Depending 
+on complexity of the final product this step is optional.
+- **Release** (optional)  
+In CI implementations releasing product typically starts and announces applications on a development 
+server used for extended functionality testing. An extension of this, Continuous Delivery (CD), 
+automates functionality testing and deploys to multiple [release runway](https://www.techtarget.com/searchsoftwarequality/definition/staging-environment) 
+destinations, including potentially directly to production. Releasing compiled product is optional 
+depending on intended use cases (executable vs. API library) or whether further testing is desired.
+
+### Actors and Automation
+A more robust view of CI architecture describes the participants, both users and automated systems. 
+The CI process always starts with developers who commit code, and it finishes with reliable, 
+unchanging, versioned and packaged software delivered by an automated system.  
+```mermaid
+flowchart TB
+    D1([fa:fa-desktop Dev 1]) -->|push| R[fa:fa-server Version Control Server]
+    D2([fa:fa-desktop Dev 2]) -->|push| R
+    Dn([fa:fa-desktop Dev n]) -->|push| R
+    R -->|trigger| C[fa:fa-server CI Server]
+    C-->|validate| R
+    C-->|report| RS[fa:fa-server Reporting Service]
+    Q{"Compile<br/>& Test"}
+    S((Success))
+    F[Notify]
+    style Q fill:#FFAC1C,stroke:#333,stroke-width:4px
+    C-->Q
+    Q-->S
+    Q-->|exception<br/>or change|F
+    S-->|publish| AR[fa:fa-server Artifact Repository]
+    S-.->|"deploy (opt.)"| AS[fa:fa-server Application Server]
+    subgraph Community
+        U1([fa:fa-user User 1])
+        U2([fa:fa-user User 2])
+        Un([fa:fa-users Users n])
+    end
+    style Community fill:#ECFFDC,stroke:#333,stroke-width:2px
+    AR<-->U1 & U2
+    AS<-.->U2
+    RS<-.->Un
+    F-.->U2
+    style U1 fill:#F6F5F3,stroke:#333,stroke-width:1px
+    style U2 fill:#F6F5F3,stroke:#333,stroke-width:1px
+    style Un fill:#F6F5F3,stroke:#333,stroke-width:1px
+```
+#### Participant Touchpoints
+The actors in the CI process are code maintainers and end users. The automation takes place in 
+servers or services between these actors. A more detailed understanding of the process itemizes 
+the roles and responsibilities of each participant.  
+
+:information_source: [**Action:** Push] **Developers**, or code maintainers, ("Dev 1...n") start 
```

Comment:
It was a little hard to map the diagram to these participant touchpoints you're describing. Maybe use numbers like you did with the earlier diagram within the flowchart to map to the action steps you're describing?

Reply:
Added numbering.